src/Models/ElementalArea.php

```diff
@@ -87,6 +88,41 @@ public function forTemplate()
         return $this->renderWith(static::class);
     }
 
+    /**
+     * @param ArrayList $elements
+     * @return $this
+     */
+    public function setElementsCached(ArrayList $elements)
+    {
+        $this->cacheData['elements'] = $elements;
+
+        return $this;
+    }
+
+    /**
+     * @param DataObject $page
+     * @return $this
+     */
+    public function setOwnerPageCached(DataObject $page)
+    {
+        $this->cacheData['owner_page'] = $page;
+
+        return $this;
+    }
+
+    /**
+     * A cache-aware accessor for the elements
+     * @return ArrayList|DataList
```

Comment:
Could you keep the existing type hint? So make it `ArrayList|HasManyList|BaseElement[]`?

Reply:
PSR-5 (draft) is introducing type collections 😃 `ArrayList<BaseElement>`

Reply:
To this effect you can also remove the `@method` hint at the top.